en/docs/identity-providers/exchange-an-access-token-from-an-external-identity-provider.md

```diff
@@ -0,0 +1,43 @@
+# Exchange an Access Token from an External Identity Provider (IdP)
+
+After you [Register an External Identity Provider]({{base_path}}/identity-providers/register-an-external-identity-provider) you can exchange access tokens from that IdP for Choreo access tokens and use those to invoke APIs. 
+
+Follow this procedure to exchange an access token from an external IdP for a Choreo token within your organization.
+
+1. Sign in to the Choreo Developer Portal at [https://devportal.choreo.dev/](https://devportal.choreo.dev/).
+
+2. Click **Applications** tab.
+
+3. Create a new application or use an existing application.
+
+4. Select **OAuth2 Tokens** under the **Production Keys** on the left bar of the application.
+
+5. If the Consumer Key and the Consumer Secret have not been generated before, the following page will be displayed. Click on **Generate Keys** to generate the Consumer Key and the Consumer Secret.
```

Comment:
Based on the latest UI, seems like "Generate Keys" has to be changed to "Generate Credentials". Please confirm.

Reply:
+1